proto-lens-protoc/src/Data/ProtoLens/Setup.hs

```diff
@@ -209,10 +216,41 @@ generateSources root l files = do
                      [ InstalledPackageInfo.dataDir info </> protoLensImportsPrefix
                      | info <- collectDeps l
                      ]
-    generateProtosWithImports (root : importDirs) (autogenModulesDir l)
+    -- Generate .hs files into a temporary directory, then move them over
+    -- to the target (autogen) directory only if they are different from
+    -- what's already there. This way, we don't needlessly touch the generated
+    -- .hs files when nothing changes, and thus don't needlessly make GHC
+    -- recompile them (as it considers their modification times for that).
+    -- See https://github.com/google/proto-lens/issues/176
```

Comment:
Style nit: My preference is to avoid mentioning the issue number in the comment; it's redundant since we can rely on the other text (which you have written well) and/or on `git blame` for future readers to track this code back to its creation.

If you would still rather keep the issue number, that's fine, just reference it as `#176` rather than the full URL.